File: source/clibCore/Interfaces/cpApi/CPSiteClass.vb

Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
Imports System.Windows.Forms
Imports Contensive.BaseClasses
Imports System.Runtime.InteropServices

Namespace Contensive.Core
    '
    ' comVisible to be activeScript compatible
    '
    <ComVisible(True)>
    <ComClass(CPSiteClass.ClassId, CPSiteClass.InterfaceId, CPSiteClass.EventsId)>
    Public Class CPSiteClass
        Inherits BaseClasses.CPSiteBaseClass
        Implements IDisposable
        '
#Region "COM GUIDs"
        Public Const ClassId As String = "7C159DA2-6677-426B-8631-3F235F24BCF0"
        Public Const InterfaceId As String = "50DD3209-AE54-46EF-8344-C6CD8960DD65"
        Public Const EventsId As String = "5E88DB23-E8D7-4CE8-9793-9C7A20F4CF3A"
#End Region
        '
        Private cpCore As Contensive.Core.coreClass
        Private CP As CPClass
        Protected disposed As Boolean = False
        '
        Public Sub New(ByVal cpCoreObj As Contensive.Core.coreClass, ByRef CPParent As CPClass)
            MyBase.New()
            cpCore = cpCoreObj
            CP = CPParent
        End Sub
        '
        ' dispose
        '
        Protected Overridable Overloads Sub Dispose(ByVal disposing As Boolean)
            If Not Me.disposed Then
                Call appendDebugLog(".dispose, dereference cp, main, csv")
                If disposing Then
                    '
                    ' call .dispose for managed objects
                    '
                    CP = Nothing
                    cpCore = Nothing
                End If
                '
                ' Add code here to release the unmanaged resource.
                '
            End If
            Me.disposed = True
        End Sub
        '
        '
        '
        Public Overrides ReadOnly Property Name() As String 'Inherits BaseClasses.CPSiteBaseClass.Name
            Get
                Return cpCore.serverConfig.appConfig.name
            End Get
        End Property
        '
        '
        '
        Public Overrides Sub SetProperty(ByVal FieldName As String, ByVal FieldValue As String) 'Inherits BaseClasses.CPSiteBaseClass.SetProperty
            Call cpCore.siteProperties.setProperty(FieldName, FieldValue)
        End Sub
        '
        '
        '
        Public Overrides Function GetProperty(ByVal FieldName As String, Optional ByVal DefaultValue As String = "") As String 'Inherits BaseClasses.CPSiteBaseClass.GetProperty
            Return cpCore.siteProperties.getText(FieldName, DefaultValue)
        End Function
        '
        '=======================================================================================================
        '
        '=======================================================================================================
        '
        Public Overrides Function GetBoolean(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Boolean
            Return genericController.EncodeBoolean(GetProperty(PropertyName, DefaultValue))
        End Function
        '
        '=======================================================================================================
        '
        '=======================================================================================================
        '
        Public Overrides Function GetDate(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Date
            Return genericController.EncodeDate(GetProperty(PropertyName, DefaultValue))
        End Function
        '
        '=======================================================================================================
        '
        '=======================================================================================================
        '
        Public Overrides Function GetInteger(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Integer
            Return CP.Utils.EncodeInteger(GetProperty(PropertyName, DefaultValue))
        End Function
        '
        '=======================================================================================================
        '
        '=======================================================================================================
        '
        Public Overrides Function GetNumber(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Double
            Return CP.Utils.EncodeNumber(GetProperty(PropertyName, DefaultValue))
        End Function
        '
        '=======================================================================================================
        '
        '=======================================================================================================
        '
        Public Overrides Function GetText(ByVal FieldName As String, Optional ByVal DefaultValue As String = "") As String
            Return GetProperty(FieldName, DefaultValue)
        End Function
        '
        '
        '
        Public Overrides ReadOnly Property MultiDomainMode() As Boolean 'Inherits BaseClasses.CPSiteBaseClass.MultiDomainMode
            Get
                MultiDomainMode = False
                If genericController.vbInstr(1, "," & cpCore.app_domainList & ",", ",*,", vbTextCompare) <> 0 Then
                    MultiDomainMode = True
                End If
            End Get
        End Property
        '
        '
        '
        <Obsolete("Deprecated, please use cp.File.cdnFiles, cp.File.privateFiles, cp.File.appRootFiles, or cp.File.serverFiles instead.", True)>
        Public Overrides ReadOnly Property PhysicalFilePath() As String 'Inherits BaseClasses.CPSiteBaseClass.PhysicalFilePath
            Get
                'cpCore.handleException(New ApplicationException("PhysicalFilePath is no longer supported. Use cp.file.cdnFiles instead to support scale modes"))
                Return cpCore.cdnFiles.rootLocalPath
            End Get
        End Property
        '
        '
        '
        <Obsolete("Deprecated, please use cp.File.cdnFiles, cp.File.privateFiles, cp.File.appRootFiles, or cp.File.serverFiles instead.", True)>
        Public Overrides ReadOnly Property PhysicalInstallPath() As String 'Inherits BaseClasses.CPSiteBaseClass.PhysicalInstallPath
            Get
                'cpCore.handleException(New ApplicationException("physicalInstallPath is no longer supported"))
                Return cpCore.privateFiles.rootLocalPath
            End Get
        End Property
        '
        '
        '
        <Obsolete("Deprecated, please use cp.File.cdnFiles, cp.File.privateFiles, cp.File.appRootFiles, or cp.File.serverFiles instead.", True)>
        Public Overrides ReadOnly Property PhysicalWWWPath() As String 'Inherits BaseClasses.CPSiteBaseClass.PhysicalWWWPath
            Get
                'cpCore.handleException(New ApplicationException("PhysicalFilePath is no longer supported. Use cp.file.appRootFiles instead to support scale modes"))
                Return cpCore.appRootFiles.rootLocalPath
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property TrapErrors() As Boolean 'Inherits BaseClasses.CPSiteBaseClass.TrapErrors
            Get
                Return genericController.EncodeBoolean(GetProperty("TrapErrors", "1"))
            End Get
        End Property

        Public Overrides ReadOnly Property AppPath() As String 'Inherits BaseClasses.CPSiteBaseClass.AppPath
            Get
                Return AppRootPath
                'Return cpCore.web_requestAppPath
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property AppRootPath() As String 'Inherits BaseClasses.CPSiteBaseClass.AppRootPath
            Get
                If False Then
                    Return "/"
                Else
                    Return requestAppRootPath
                End If
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property DomainPrimary() As String 'Inherits BaseClasses.CPSiteBaseClass.DomainPrimary
            Get
                DomainPrimary = ""
                If cpCore.serverConfig.appConfig.domainList.Count > 0 Then
                    DomainPrimary = cpCore.serverConfig.appConfig.domainList(0)
                End If
                Return DomainPrimary
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property Domain() As String 'Inherits BaseClasses.CPSiteBaseClass.Domain
            Get
                If False Then
                    Return DomainPrimary
                Else
                    Return cpCore.webServer.webServerIO_requestDomain
                End If
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property DomainList() As String 'Inherits BaseClasses.CPSiteBaseClass.DomainList
            Get
                Return cpCore.app_domainList
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property FilePath() As String 'Inherits BaseClasses.CPSiteBaseClass.FilePath
            Get
                Return cpCore.serverConfig.appConfig.cdnFilesNetprefix ' "/" & cpCore.app.config.name & "/files/"
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property PageDefault() As String 'Inherits BaseClasses.CPSiteBaseClass.PageDefault
            Get
                If False Then
                    Return GetProperty(siteproperty_serverPageDefault_name, "")
                Else
                    Return cpCore.siteProperties.serverPageDefault
                End If
            End Get
        End Property
        '
        '
        '
        Public Overrides ReadOnly Property VirtualPath() As String 'Inherits BaseClasses.CPSiteBaseClass.VirtualPath
            Get
                Return "/" & cpCore.serverConfig.appConfig.name
            End Get
        End Property
        '
        '
        '
        Public Overrides Function EncodeAppRootPath(ByVal Link As String) As String 'Inherits BaseClasses.CPSiteBaseClass.EncodeAppRootPath
            If False Then
                Return Link
            Else
                Return genericController.EncodeAppRootPath(genericController.encodeText(Link), cpCore.webServer.webServerIO_requestVirtualFilePath, requestAppRootPath, cpCore.webServer.requestDomain)
            End If
        End Function

        Public Overrides Function IsTesting() As Boolean 'Inherits BaseClasses.CPSiteBaseClass.IsTesting
            Return False
        End Function

        Public Overrides Sub LogActivity(ByVal Message As String, ByVal UserID As Integer, ByVal OrganizationID As Integer) 'Inherits BaseClasses.CPSiteBaseClass.LogActivity
            Call cpCore.log_logActivity(Message, 0, UserID, OrganizationID)
        End Sub
        '
        ' Report an alarm
        '   This will set off the server monitor
        '
        Public Overrides Sub LogWarning(ByVal name As String, ByVal description As String, ByVal typeOfWarningKey As String, ByVal instanceKey As String)
            Call cpCore.csv_reportWarning(name, description, typeOfWarningKey, instanceKey)
        End Sub
        '
        ' Report an alarm
        '   This will set off the server monitor
        '
        Public Overrides Sub LogAlarm(ByVal cause As String)
            Call cpCore.csv_reportAlarm(cause)
        End Sub
        '
        ' Report an error message
        '   Eventually, move the cmc.main_ReportError code here so process errors get emailed, etc.
        '
        Public Overrides Sub ErrorReport(ByVal Cause As String) 'Inherits BaseClasses.CPSiteBaseClass.ErrorReport
            '
            throw (New ApplicationException("Unexpected exception")) 'cpCore.handleLegacyError8(Cause, "", True)
        End Sub
        '
        ' Report a structured exception object event, with a message
        '
        Public Overrides Sub ErrorReport(ByVal Ex As System.Exception, Optional ByVal Message As String = "") 'Inherits BaseClasses.CPSiteBaseClass.ErrorReport
            If Message = "" Then
                Call cpCore.handleException(Ex, "n/a", 2)
            Else
                Call cpCore.handleException(Ex, Message, 2)
            End If
            'Dim s As String = ""
            's = Ex.Source & ", " & Ex.ToString
            'If Message <> "" Then
            '    s = Message & " [" & s & "]"
            'End If
            'cpCore.handleLegacyError8(s, "", True)
        End Sub

        Public Overrides Sub RequestTask(ByVal Command As String, ByVal SQL As String, ByVal ExportName As String, ByVal Filename As String) 'Inherits BaseClasses.CPSiteBaseClass.RequestTask
            If False Then
                Call cpCore.tasks_RequestTask(Command, SQL, ExportName, Filename, 0)
            Else
                Call cpCore.main_RequestTask(Command, SQL, ExportName, Filename)
            End If
        End Sub

        Public Overrides Sub TestPoint(ByVal Message As String) 'Inherits BaseClasses.CPSiteBaseClass.TestPoint
            If False Then
                '
                '
                '
            Else
                Call cpCore.debug_testPoint(Message)
            End If
        End Sub

        Public Overrides Function LandingPageId(Optional ByVal DomainName As String = "") As Integer 'Inherits BaseClasses.CPSiteBaseClass.LandingPageId
            Return GetProperty("LandingPageID", "")
        End Function
        '
        '
        '
        Public Overrides Sub addLinkAlias(ByVal linkAlias As String, ByVal pageId As Integer, Optional ByVal queryStringSuffix As String = "")
            Call cpCore.app_addLinkAlias(linkAlias, pageId, queryStringSuffix)
        End Sub
        '
        '
        '
        Public Overrides Function throwEvent(ByVal eventNameIdOrGuid As String) As String
            Dim returnString As String = ""
            Try
                Dim sql As String
                Dim cs As CPCSClass = CP.CSNew()
                Dim addonid As Integer = 0
                '
                sql = "select e.id,c.addonId" _
                    & " from (ccAddonEvents e" _
                    & " left join ccAddonEventCatchers c on c.eventId=e.id)" _
                    & " where "
                If genericController.vbIsNumeric(eventNameIdOrGuid) Then
                    sql &= "e.id=" & CP.Db.EncodeSQLNumber(CDbl(eventNameIdOrGuid))
                ElseIf CP.Utils.isGuid(eventNameIdOrGuid) Then
                    sql &= "e.ccGuid=" & CP.Db.EncodeSQLText(eventNameIdOrGuid)
                Else
                    sql &= "e.name=" & CP.Db.EncodeSQLText(eventNameIdOrGuid)
                End If
                If Not cs.OpenSQL(sql) Then
                    '
                    ' event not found
                    '
                    If genericController.vbIsNumeric(eventNameIdOrGuid) Then
                        '
                        ' can not create an id
                        '
                    ElseIf CP.Utils.isGuid(eventNameIdOrGuid) Then
                        '
                        ' create event with Guid and id for name
                        '
                        Call cs.Close()
                        Call cs.Insert("add-on Events")
                        Call cs.SetField("ccguid", eventNameIdOrGuid)
                        Call cs.SetField("name", "Event " & cs.GetInteger("id").ToString())
                    ElseIf (eventNameIdOrGuid <> "") Then
                        '
                        ' create event with name
                        '
                        Call cs.Close()
                        Call cs.Insert("add-on Events")
                        Call cs.SetField("name", eventNameIdOrGuid)
                    End If
                Else
                    Do While cs.OK
                        addonid = cs.GetInteger("addonid")
                        If addonid <> 0 Then
                            returnString &= CP.Utils.ExecuteAddon(addonid)
                        End If
                        Call cs.GoNext()
                    Loop
                End If
                Call cs.Close()
                '
            Catch ex As Exception
                Call ErrorReport(ex, "Expected error in throwEvent()")
            End Try
            Return returnString
        End Function
        ''
        '' 20151121 removed to resolve issue with com compatibility
        '' Report an unstructured Err object event, with a message
        ''
        'Public Overrides Sub ErrorReport(ByVal Err As Microsoft.VisualBasic.ErrObject, Optional ByVal Message As String = "") 'Inherits BaseClasses.CPSiteBaseClass.ErrorReport
        '    Dim Ex As System.Exception
        '    '
        '    Ex = Err.GetException
        '    CP.Utils.AppendLog(Message & vbCrLf & "source " & Err.Source & vbCrLf & "#" & Err.Number & vbCrLf & "line " & Err.Erl & vbCrLf & "description: " & Err.Description & "]" & vbCrLf & Ex.ToString)
        '    If True Then
        '        '
        '        ' if main is available, report it out that way too
        '        '
        '        throw new applicationException("Unexpected exception") ' Call cpcore.handleLegacyError23(Message)
        '    End If
        'End Sub
        '
        '
        '
        Private Sub appendDebugLog(ByVal copy As String)
            'My.Computer.FileSystem.WriteAllText("c:\clibCpDebug.log", Now & " - cp.site, " & copy & vbCrLf, True)
            ' 'My.Computer.FileSystem.WriteAllText(System.AppDocmc.main_CurrentDocmc.main_BaseDirectory() & "cpLog.txt", Now & " - " & copy & vbCrLf, True)
        End Sub
        '
        ' testpoint
        '
        Private Sub tp(ByVal msg As String)
            'Call appendDebugLog(msg)
        End Sub
#Region " IDisposable Support "
        ' Do not change or add Overridable to these methods.
        ' Put cleanup code in Dispose(ByVal disposing As Boolean).
        Public Overloads Sub Dispose() Implements IDisposable.Dispose
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
        Protected Overrides Sub Finalize()
            Dispose(False)
            MyBase.Finalize()
        End Sub
#End Region
    End Class
End Namespace

File: source/clibCoreCs/Interfaces/cpApi/CPSiteClass.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
//
namespace Contensive.Core {
    public class CPSiteClass : BaseClasses.CPSiteBaseClass, IDisposable {
        //
        #region COM GUIDs
        public const string ClassId = "7C159DA2-6677-426B-8631-3F235F24BCF0";
        public const string InterfaceId = "50DD3209-AE54-46EF-8344-C6CD8960DD65";
        public const string EventsId = "5E88DB23-E8D7-4CE8-9793-9C7A20F4CF3A";
        #endregion
        //
        private Contensive.Core.coreClass cpCore;
        private CPClass CP;
        protected bool disposed = false;
        //
        //====================================================================================================
        //
        public CPSiteClass(Contensive.Core.coreClass cpCoreObj, CPClass CPParent) : base() {
            cpCore = cpCoreObj;
            CP = CPParent;
        }
        //
        //====================================================================================================
        //
        protected virtual void Dispose(bool disposing) {
            if (!this.disposed) {
                if (disposing) {
                    //
                    // call .dispose for managed objects
                    //
                    CP = null;
                    cpCore = null;
                }
                //
                // Add code here to release the unmanaged resource.
                //
            }
            this.disposed = true;
        }
        //
        //====================================================================================================
        //
        public override string Name {
            get {
                return cpCore.serverConfig.appConfig.name;
            }
        }
        //
        //====================================================================================================
        //
        public override void SetProperty(string FieldName, string FieldValue) {
            cpCore.siteProperties.setProperty(FieldName, FieldValue);
        }
        //
        //
        //
        public override string GetProperty(string FieldName, string DefaultValue = "") {
            return cpCore.siteProperties.getText(FieldName, DefaultValue);
        }
        //
        //====================================================================================================
        //
        public override bool GetBoolean(string PropertyName, string DefaultValue = "") {
            return genericController.EncodeBoolean(GetProperty(PropertyName, DefaultValue));
        }
        //
        //====================================================================================================
        //
        public override DateTime GetDate(string PropertyName, string DefaultValue = "") {
            return genericController.EncodeDate(GetProperty(PropertyName, DefaultValue));
        }
        //
        //====================================================================================================
        //
        public override int GetInteger(string PropertyName, string DefaultValue = "") {
            return CP.Utils.EncodeInteger(GetProperty(PropertyName, DefaultValue));
        }
        //
        //====================================================================================================
        //
        public override double GetNumber(string PropertyName, string DefaultValue = "") {
            return CP.Utils.EncodeNumber(GetProperty(PropertyName, DefaultValue));
        }
        //
        //====================================================================================================
        //
        public override string GetText(string FieldName, string DefaultValue = "") {
            return GetProperty(FieldName, DefaultValue);
        }
        //
        //====================================================================================================
        //
        public override bool MultiDomainMode {
            get {
                return false;
            }
        }
        //
        //====================================================================================================
        //
        [Obsolete("Deprecated, please use cp.File.cdnFiles, cp.File.privateFiles, cp.File.appRootFiles, or cp.File.serverFiles instead.", true)]
        public override string PhysicalFilePath {
            get {
                //cpCore.handleException(New ApplicationException("PhysicalFilePath is no longer supported. Use cp.file.cdnFiles instead to support scale modes"))
                return cpCore.cdnFiles.rootLocalPath;
            }
        }
        //
        //====================================================================================================
        //
        [Obsolete("Deprecated, please use cp.File.cdnFiles, cp.File.privateFiles, cp.File.appRootFiles, or cp.File.serverFiles instead.", true)]
        public override string PhysicalInstallPath {
            get {
                //cpCore.handleException(New ApplicationException("physicalInstallPath is no longer supported"))
                return cpCore.privateFiles.rootLocalPath;
            }
        }
        //
        //====================================================================================================
        //
        [Obsolete("Deprecated, please use cp.File.cdnFiles, cp.File.privateFiles, cp.File.appRootFiles, or cp.File.serverFiles instead.", true)]
        public override string PhysicalWWWPath {
            get {
                //cpCore.handleException(New ApplicationException("PhysicalFilePath is no longer supported. Use cp.file.appRootFiles instead to support scale modes"))
                return cpCore.appRootFiles.rootLocalPath;
            }
        }
        //
        //====================================================================================================
        //
        public override bool TrapErrors {
            get {
                return genericController.EncodeBoolean(GetProperty("TrapErrors", "1"));
            }
        }
        //
        //====================================================================================================
        //
        public override string AppPath {
            get {
                return AppRootPath;
                //Return cpCore.web_requestAppPath
            }
        }
        //
        //====================================================================================================
        //
        public override string AppRootPath //Inherits BaseClasses.CPSiteBaseClass.AppRootPath
        {
            get {
                if (false) {
                    return "/";
                } else {
                    return requestAppRootPath;
                }
            }
        }
        //
        //====================================================================================================
        //
        public override string DomainPrimary //Inherits BaseClasses.CPSiteBaseClass.DomainPrimary
        {
            get {
                string tempDomainPrimary = null;
                tempDomainPrimary = "";
                if (cpCore.serverConfig.appConfig.domainList.Count > 0) {
                    tempDomainPrimary = cpCore.serverConfig.appConfig.domainList[0];
                }
                return tempDomainPrimary;
            }
        }
        //
        //====================================================================================================
        //
        public override string Domain //Inherits BaseClasses.CPSiteBaseClass.Domain
        {
            get {
                if (false) {
                    return DomainPrimary;
                } else {
                    return cpCore.webServer.requestDomain;
                }
            }
        }
        //
        //====================================================================================================
        //
        public override string DomainList //Inherits BaseClasses.CPSiteBaseClass.DomainList
        {
            get {
                return string.Join(",", cpCore.serverConfig.appConfig.domainList);
            }
        }
        //
        //====================================================================================================
        //
        public override string FilePath //Inherits BaseClasses.CPSiteBaseClass.FilePath
        {
            get {
                return cpCore.serverConfig.appConfig.cdnFilesNetprefix; // "/" & cpCore.app.config.name & "/files/"
            }
        }
        //
        //====================================================================================================
        //
        public override string PageDefault //Inherits BaseClasses.CPSiteBaseClass.PageDefault
        {
            get {
                if (false) {
                    return GetProperty(siteproperty_serverPageDefault_name, "");
                } else {
                    return cpCore.siteProperties.serverPageDefault;
                }
            }
        }
        //
        //====================================================================================================
        //
        public override string VirtualPath //Inherits BaseClasses.CPSiteBaseClass.VirtualPath
        {
            get {
                return "/" + cpCore.serverConfig.appConfig.name;
            }
        }
        //
        //====================================================================================================
        //
        public override string EncodeAppRootPath(string Link) //Inherits BaseClasses.CPSiteBaseClass.EncodeAppRootPath
        {
            if (false) {
                return Link;
            } else {
                return genericController.EncodeAppRootPath(genericController.encodeText(Link), cpCore.webServer.requestVirtualFilePath, requestAppRootPath, cpCore.webServer.requestDomain);
            }
        }
        //
        //====================================================================================================
        //
        public override bool IsTesting() //Inherits BaseClasses.CPSiteBaseClass.IsTesting
        {
            return false;
        }
        //
        //====================================================================================================
        //
        public override void LogActivity(string Message, int UserID, int OrganizationID) //Inherits BaseClasses.CPSiteBaseClass.LogActivity
        {
            logController.logActivity(cpCore, Message, 0, UserID, OrganizationID);
        }
        //
        //====================================================================================================
        //
        public override void LogWarning(string name, string description, string typeOfWarningKey, string instanceKey) {
            logController.csv_reportWarning(cpCore, name, description, "", 0, description, typeOfWarningKey, instanceKey);
        }
        //
        //====================================================================================================
        //
        public override void LogAlarm(string cause) {
            logController.appendLog(cpCore, cause, "Alarm");
        }
        //
        //====================================================================================================
        //
        public override void ErrorReport(string Cause) //Inherits BaseClasses.CPSiteBaseClass.ErrorReport
        {
            //
            throw (new ApplicationException("Unexpected exception")); //cpCore.handleLegacyError8(Cause, "", True)
        }
        //
        //====================================================================================================
        //
        public override void ErrorReport(System.Exception Ex, string Message = "") //Inherits BaseClasses.CPSiteBaseClass.ErrorReport
        {
            if (string.IsNullOrEmpty(Message)) {
                cpCore.handleException(Ex, "n/a", 2);
            } else {
                cpCore.handleException(Ex, Message, 2);
            }
            //Dim s As String = ""
            //s = Ex.Source & ", " & Ex.ToString
            //If Message <> "" Then
            //    s = Message & " [" & s & "]"
            //End If
            //cpCore.handleLegacyError8(s, "", True)
        }
        //
        //====================================================================================================
        //
        public override void RequestTask(string Command, string SQL, string ExportName, string Filename) //Inherits BaseClasses.CPSiteBaseClass.RequestTask
        {
            taskSchedulerController.main_RequestTask(cpCore, Command, SQL, ExportName, Filename);
        }
        //
        //====================================================================================================
        //
        public override void TestPoint(string Message) //Inherits BaseClasses.CPSiteBaseClass.TestPoint
        {
            debugController.testPoint(cpCore, Message);
        }
        //
        //====================================================================================================
        //
        public override int LandingPageId(string DomainName = "") //Inherits BaseClasses.CPSiteBaseClass.LandingPageId
        {
            return Convert.ToInt32(GetProperty("LandingPageID", ""));
        }
        //
        //====================================================================================================
        //
        public override void addLinkAlias(string linkAlias, int pageId, string queryStringSuffix = "") {
            docController.addLinkAlias(cpCore, linkAlias, pageId, queryStringSuffix);
        }
        //
        //====================================================================================================
        //
        public override string ThrowEvent(string eventNameIdOrGuid) {
            return cpCore.addon.throwEvent(eventNameIdOrGuid);
        }
        //
        //====================================================================================================
        //
        public override Dictionary<string, routeClass> getRouteDictionary() {
            return cpCore.routeDictionary;
        }

        #region  IDisposable Support 
        // Do not change or add Overridable to these methods.
        // Put cleanup code in Dispose(ByVal disposing As Boolean).
        public void Dispose() {
            Dispose(true);
            GC.SuppressFinalize(this);
        }

        ~CPSiteClass() {
            Dispose(false);
        }
        #endregion
    }
}